src/app/ReadHandler.h

```diff
@@ -0,0 +1,161 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *     This file defines read handler for a CHIP Interaction Data model
+ *
+ */
+
+#pragma once
+
+#ifndef _CHIP_INTERACTION_MODEL_READ_HANDLER_H
+#define _CHIP_INTERACTION_MODEL_READ_HANDLER_H
+
+#include <app/reporting/EventLoggingTypes.h>
+#include <app/reporting/LoggingManagement.h>
+#include <core/CHIPCore.h>
+#include <core/CHIPTLVDebug.hpp>
+#include <map>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace app {
+namespace reporting {
+class ReportingEngine;
+}
+class ReadHandler
+{
+public:
+    enum EventID
+    {
+        kEvent_OnReadRequestParsed = 0,
+    };
+
+    union InEventParam
+    {
+        void Clear(void) { memset(this, 0, sizeof(*this)); }
+
+        struct
+        {
+            bool mGetToAllEvents;
+            // Do not close the EC
```

Comment:
similar comments to previous class:
  - please expand comments and use `const` instead of a do no modify comment
  - use full names (exchangeContext instead of `EC`)

Reply:
Done, thanks